internal/exec/stages/files/files_test.go

```diff
@@ -19,56 +19,208 @@ import (
 
 	type out struct {
-		data []string
+		data []types.Directory
 	}
 
 	tests := []struct {
 		in  in
 		out out
 	}{
 		{
-			in:  in{data: []string{"/a/b/c/d/e/", "/a/b/c/d/", "/a/b/c/", "/a/b/", "/a/"}},
-			out: out{data: []string{"/a/", "/a/b/", "/a/b/c/", "/a/b/c/d/", "/a/b/c/d/e/"}},
+			in: in{data: []types.Directory{
+				{
+					Node: types.Node{
+						Path: "/a/b/c/d/e/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/d/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/",
+					},
+				},
+			}},
+			out: out{data: []types.Directory{
+				{
+					Node: types.Node{
+						Path: "/a/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/d/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/d/e/",
+					},
+				},
+			}},
 		},
 		{
-			in:  in{data: []string{"/a////b/c/d/e/", "/", "/a/b/c//d/", "/a/b/c/", "/a/b/", "/a/"}},
-			out: out{data: []string{"/", "/a/", "/a/b/", "/a/b/c/", "/a/b/c//d/", "/a////b/c/d/e/"}},
+			in: in{data: []types.Directory{
+				{
+					Node: types.Node{
+						Path: "/a////b/c/d/e/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c//d/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/",
+					},
+				},
+			}},
+			out: out{data: []types.Directory{
+				{
+					Node: types.Node{
+						Path: "/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/b/c//d/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a////b/c/d/e/",
+					},
+				},
+			}},
 		},
 		{
-			in:  in{data: []string{"/a/", "/a/../a/b", "/"}},
-			out: out{data: []string{"/", "/a/", "/a/../a/b"}},
+			in: in{data: []types.Directory{
+				{
+					Node: types.Node{
+						Path: "/a/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/../a/b",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/",
+					},
+				},
+			}},
+			out: out{data: []types.Directory{
+				{
+					Node: types.Node{
+						Path: "/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/",
+					},
+				},
+				{
+					Node: types.Node{
+						Path: "/a/../a/b",
+					},
+				},
+			}},
 		},
 	}
 
 	for i, test := range tests {
-		dirs := make([]pathWrapper, len(test.in.data))
+		entries := []pathWrapper{}
```

Comment:
Can we not drop `pathWrapper` (and it's interface definition) in favor of just using `filesystemEntry`?

Reply:
Yeah, makes sense! Addressed this in the latesh push.